lib/ex_auction/auction/worker.ex

```diff
@@ -125,14 +123,14 @@ defmodule ExAuction.Auction.Worker do
   end
 
   defp whereis(auction) do
-    case :global.whereis_name(%Auction{auction | status: nil, pid: nil}) do
+    case :global.whereis_name(auction.name) do
```

Comment:
since we are now using auction name as key in registry, `whereis/1` should accept just name something like

```
 defp whereis(name) do
    case :global.whereis_name(name) do
    ...
```